magma/placer.py

```diff
@@ -129,6 +129,9 @@ def place(self, inst):
             inst.stack = inspect.stack()
         _setup_view(inst)
 
+    def finalize(self, defn):
```

Comment:
ik functionally it doesn't matter, but let's add the same `if self._finalized: raise Exception()` logic